### PR TITLE
HOCS-2643: refactor deploy script

### PR DESCRIPTION
**Pull request by LiamMacP**
This refactor contains the following changes:
- Pull most of the exports to the top of the file
- Remove the need to specify domain - this change works out if `wcs` or
`wcs`
- Use `VERSION` for build instead of `IMAGE_VERSION`
- Remove unnecessary `KUBE_TOKEN` if/else tree

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -22,41 +21,7 @@ else
...
-    export KUBE_TOKEN=${HOCS_FRONTEND_QAX}
-else
-    echo "Unable to find environment: ${ENVIRONMENT}"
-fi
-    
-if [[ -z ${KUBE_TOKEN} ]] ; then
-    echo "Failed to find a value for KUBE_TOKEN - exiting"
-    exit -1
-fi
+export KC_REALM=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hocs-notprod
```
Comment (zuzak) [inline deploy.sh:24]: Move this up two lines?

File: deploy.sh [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-if [[ -z ${VERSION} ]] ; then
-    export VERSION=${IMAGE_VERSION}
-fi
-if [[ -z ${DOMAIN} ]] ; then
-    export DOMAIN="cs"
+export KUBE_TOKEN=${KUBE_TOKEN}
+export VERSION=${VERSION}
+  
+export DOMAIN="cs"
+if [ $(echo ${KUBE_NAMESPACE} | sed 's/-.*//') == "wcs" ]; then
```
Comment (zuzak) [inline deploy.sh:11]: All those Drone string manipulation functions are lifted from Bash, which means we can do:
```suggestion
if [ ${KUBE_NAMESPACE%-*} == "wcs" ]; then
```

File: deploy.sh [hunk 3]
```diff
@@ -1,17 +1,16 @@
 #!/bin/bash
+set -euo pipefail
 
+export IP_WHITELIST=${POISE_WHITELIST}
 export KUBE_NAMESPACE=${ENVIRONMENT}
 export KUBE_SERVER=${KUBE_SERVER}
```
Comment (zuzak) [inline deploy.sh:6]: We should construct `KUBE_SERVER` based on namespace too. But that can be a later commit, along with removing `--insecure-skip-tls-verify`.

**Review by zuzak — APPROVED** [review]

File: deploy.sh [hunk 4]
```diff
@@ -66,37 +31,28 @@ elif [[ "${KUBE_NAMESPACE}" == "cs-prod" ]] ; then
...
     export DNS_PREFIX=demo.cs-notprod
-    export KC_REALM=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hocs-notprod
 elif [[ "${KUBE_NAMESPACE}" == "wcs-demo" ]] ; then
     export DNS_PREFIX=demo.wcs-notprod
-    export KC_REALM=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hocs-notprod
 elif [[ "${KUBE_NAMESPACE}" == "hocs-qax" ]] ; then
     export DNS_PREFIX=qax.internal.cs-notprod
-    export KC_REALM=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hocs-notprod
 else
     export DNS_PREFIX=${ENVIRONMENT}.${DOMAIN}-notprod
```
Comment (zuzak) [inline deploy.sh:47]: This should probably default to `internal`, now that we're actually gonna use this?

File: deploy.sh (hunk 1 above)
Comment (LiamMacP) [reply deploy.sh]: Done 👍🏼

File: deploy.sh (hunk 2 above)
Comment (LiamMacP) [reply deploy.sh]: Done 👍🏼

File: deploy.sh (hunk 3 above)
Comment (LiamMacP) [reply deploy.sh:6]: That can be another commit - you have 2 open PR's to optimise this further so will look at that at a later date 👍🏼

File: deploy.sh (hunk 4 above)
Comment (LiamMacP) [reply deploy.sh]: Done 👍🏼

**Pull request merged**